### PR TITLE
[charts] Fix `ChartsTooltip` component setup

### LINKED ISSUE #11150: [charts] Should tooltipClasses be exported similar to axisClasses?
### Summary 💡

Similar to the [axisClasses](https://mui.com/x/react-charts/bars/#using-a-dataset) example in the documentation, changing the styling of the tooltip should be able via tooltipClasses.

I tried using `'& .MuiChartsTooltip-markCell' ` for styling the tooltip, but the changes did not apply.

Adding `export * from './tooltipClasses';` to the index.tsx in ChartsTooltip should be enough.

### Examples 🌈

_No response_

### Motivation 🔦

I would like to change to styling of the charts tooltip, for example hiding the markCell.

**Search keywords**: tooltipClasses

### PR DESCRIPTION
**Pull request by LukasTy**
Fixes #11150 

- Add the missing `chartstooltipClasses` export
- Fix for `defaultProps` in theme to work for `MuiChartsTooltip`
- Add `styleOverrides` in theme for `MuiChartsTooltip`
- Add `table`, `row`, `mark`, and `cell` slots/classes 
- Fix `getTooltipHasData` typo

### CODE REVIEW

File: packages/x-charts/src/ChartsTooltip/utils.tsx [hunk 1]
```diff
@@ -72,7 +72,7 @@ export function useMouseTracker() {
 
 export type TriggerOptions = 'item' | 'axis' | 'none';
 
-export function getTootipHasData(
+export function getTooltipHasData(
```
Comment (michelengelen) [inline packages/x-charts/src/ChartsTooltip/utils.tsx:75]: Nice catch

**Review by michelengelen — APPROVED** [review]
Looks solid! Nice work!

**Comment by LukasTy** [comment]
@michelengelen I've taken the liberty to prefix `tooltipClasses` with `charts` (https://github.com/mui/mui-x/pull/11152/commits/2eeed3abe404b2711eab80145fccfd1c99af41fc) in order to avoid the naming conflict `@mui/material/Tooltip`.
Could you have a second look and tell me if you agree with this approach? 🤔 

P.S. It wasn't previously exported, hence, it's still not a breaking change. 👌

**Review by michelengelen — APPROVED** [review]
using a prefix to differentiate is a valid approach ... so I am very much in line with what you have done @LukasTy 

I did take another look though, as you requested. Nothing to nag about with this one! Perfectly done! LGTM! :shipit:

**Pull request merged**